src/plugins/orangehrmCorePlugin/Dao/ValidationUniqueDao.php

```diff
@@ -0,0 +1,53 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA
+ */
+
+namespace OrangeHRM\Core\Dao;
+
+class ValidationUniqueDao extends BaseDao
+{
+    /**
+     * @param string $value
+     * @param string $entityName
+     * @param string $attributeName
+     * @param string|null $entityId
+     * @param string|null $entityIdField
+     * @param string|null $matchByField
+     * @param string|null $matchByValue
+     * @return bool
+     */
+    public function isValueUnique(string $value, string $entityName, string $attributeName, ?string $entityId, ?string $entityIdField, ?string $matchByField, ?string $matchByValue): bool
+    {
+        $qb = $this->createQueryBuilder('OrangeHRM\\Entity\\'  . $entityName, 'entity');
+        $qb->andWhere($qb->expr()->eq('entity.' . $attributeName, ':value'))
+            ->setParameter('value', $value);
+
+        if (!is_null($entityId)) {
+            $id = $entityIdField ?? 'id';
+            $qb->andWhere($qb->expr()->neq('entity.' . $id, ':id'))
+                ->setParameter('id', $entityId);
+        }
+
+        if (!is_null($matchByValue) && !is_null($matchByField)) {
+            $qb->andWhere($qb->expr()->eq('entity.' . $matchByField, ':matchValue'))
```

Comment:
I'm not sure this and above string concatenation to `entity.` will introduce SQL injection as `$matchByField` can be a user input according to my understanding.
you can further check https://cheatsheetseries.owasp.org/cheatsheets/SQL_Injection_Prevention_Cheat_Sheet.html

Reply:
`matchByField` is now validated in API